lib/active_merchant/billing/gateways/ipg.rb

```diff
@@ -395,6 +395,11 @@ def parse_element(reply, node)
         return reply
       end
 
+      def store_and_user_id_from(user_id)
+        split_credentials = user_id.split('._.')
+        return { store_id: split_credentials[0].sub(/^WS/, ''), user_id: split_credentials[1] }
```

Comment:
FYI `return` is implicit in most cases
```suggestion
        { store_id: split_credentials[0].sub(/^WS/, ''), user_id: split_credentials[1] }
```

Reply:
Nice thinking on accepting a string without `WS`, but we should make sure its only removed if it is the first two characters - if that's already what it does, awesome... I can't sightread regex 😄

Reply:
Removed the return! Also yes, the `^` in the regex string is a check to make sure it only matches at the beginning of the string. I included an additional test though to make sure!